tests/rptest/tests/pandaproxy_test.py

```diff
@@ -947,6 +947,86 @@ def test_consumer_group_json_v2(self):
+
+    @cluster(num_nodes=3)
+    def test_invalid_member_id(self):
+        """
+        Validates that an invalid member name is rejected
+        """
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+
+        self.logger.debug(
+            "Attempting to create consumer with a new-line character in name")
+        res = self._create_named_consumer(group_id, "my\nconsumer")
+        assert res.status_code == requests.codes.bad_request
+        assert res.json(
+        )["message"] == "consumer group name contains invalid control character"
+
+    @cluster(num_nodes=3)
+    def test_invalid_subscribe_consumer(self):
+        """
+        Validates that when subscribing to a consumer to a topic the topic name
+        is correctly validated
+        """
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+
+        self.logger.info("Creating consumer group")
+        cc_res = self._create_consumer(group_id)
+        assert cc_res.status_code == requests.codes.ok
+
+        c0 = Consumer(cc_res.json(), self.logger)
+
+        sc_res = c0.subscribe(["test\ntopic"])
+        assert sc_res.status_code == requests.codes.bad_request
+        assert sc_res.json(
+        )["message"] == "topic name contains invalid control character"
+
+    @cluster(num_nodes=3)
+    def test_invalid_topic_get_consumer_offset(self):
+        """
+        Validates that when getting consumer offsets that topic names are checked
+        for invalid control characters
+        """
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+
+        self.logger.info("Creating consumer group")
+        cc_res = self._create_consumer(group_id)
+        assert cc_res.status_code == requests.codes.ok
+
+        c0 = Consumer(cc_res.json(), self.logger)
+        co_req = dict(partitions=[dict(topic="test\ntopic", partition=0)])
+        co_res_raw = c0.get_offsets(data=json.dumps(co_req))
+        assert co_res_raw.status_code == requests.codes.bad_request
+        assert co_res_raw.json(
+        )["message"] == "requested topic name contains invalid control character"
+
+    @cluster(num_nodes=3)
+    def test_invalid_topic_commit_offset(self):
+        """
+        Validates that when committing offsets for a consumer, that the topic names
+        are checked for invalid control characters
+        """
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+
+        self.logger.info("Creating consumer group")
+        cc_res = self._create_consumer(group_id)
+        assert cc_res.status_code == requests.codes.ok
+
+        c0 = Consumer(cc_res.json(), self.logger)
+        co_req = dict(
+            partitions=[dict(topic="test\ntopic", partition=0, offset=0)])
+        co_res_raw = c0.set_offsets(data=json.dumps(co_req))
+        assert co_res_raw.status_code == requests.codes.bad_request
+        assert co_res_raw.json(
+        )["message"] == "requested topic name contains invalid control character"
+
```

Comment:
If you do add control char validation to all string types, then we need to test all handlers as well

Reply:
yeah looking forward to that ;)